engine/src/main/java/org/hibernate/validator/internal/cfg/context/DefaultConstraintMapping.java

```diff
@@ -83,4 +91,31 @@ public final AnnotationProcessingOptionsImpl getAnnotationProcessingOptions() {
 
 		return configurations;
 	}
+
+	@Override
+	public <A extends Annotation> ConstraintDefinitionContext<A> constraint(Class<A> annotationClass) {
+		Contracts.assertNotNull( annotationClass, MESSAGES.annotationTypeMustNotBeNull() );
+		Contracts.assertTrue( annotationClass.isAnnotationPresent( Constraint.class ),
```

Comment:
Ok. I suppose ideally we'd have a simple to use method `assertIsConstraintType()` which accepts an annotation type and does the validation wherever it's needed.